--- boost/libs/python/src/numpy/dtype.cpp	2022-12-08 01:02:47.000000000 +0000
+++ boost/libs/python/src/numpy/dtype.cpp.oh	2025-10-22 03:07:54.184998601 +0000
@@ -98,7 +98,14 @@
   return python::detail::new_reference(reinterpret_cast<PyObject*>(obj));
 }
 
-int dtype::get_itemsize() const { return reinterpret_cast<PyArray_Descr*>(ptr())->elsize;}
+// int dtype::get_itemsize() const { return reinterpret_cast<PyArray_Descr*>(ptr())->elsize;}
+int dtype::get_itemsize() const {
+#if NPY_ABI_VERSION < 0x02000000
+  return reinterpret_cast<PyArray_Descr*>(ptr())->elsize;
+#else
+  return PyDataType_ELSIZE(reinterpret_cast<PyArray_Descr*>(ptr()));
+#endif
+}
 
 bool equivalent(dtype const & a, dtype const & b) {
     // On Windows x64, the behaviour described on 
--- boost/libs/context/build/Jamfile.v2		2025-12-08 22:55:51.945733762 +0800
+++ boost/libs/context/build/Jamfile.v2.oh	2025-12-08 22:56:08.584375226 +0800
@@ -206,6 +206,17 @@
      <toolset>gcc
    ;
 
+alias asm_sources
+   : asm/make_arm64_aapcs_elf_gas.S
+     asm/jump_arm64_aapcs_elf_gas.S
+     asm/ontop_arm64_aapcs_elf_gas.S
+   : <abi>sysv
+     <address-model>64
+     <architecture>arm
+     <binary-format>elf
+     <toolset>gcc
+   ;
+
 # ARM64/AAPCS/MACH-O
 alias asm_sources
    : asm/make_arm64_aapcs_macho_gas.S
